Bluetooth: audio: Accept empty CCID list

MariuszSkamra opened the pull request:
The CCID list can be empty, as the Assigned Numbers is not strict regarding it's minimum length.

Fixes: #59666

MariuszSkamra committed:
Bluetooth: audio: Accept empty CCID list in audio metadata

The CCID list can be empty, as the Assigned Numbers is not strict
regarding it's minimum length.

Fixes: #59666
Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

pdgendt commented on samples/bluetooth/hap_ha/src/bap_unicast_sr.c:
```diff
@@ -297,13 +297,8 @@ static bool valid_metadata_type(uint8_t type, uint8_t len)
-		return true;
-	case BT_AUDIO_METADATA_TYPE_PROGRAM_INFO: /* 0 - 255 octets */
+	case BT_AUDIO_METADATA_TYPE_CCID_LIST:
+\	case BT_AUDIO_METADATA_TYPE_PROGRAM_INFO: /* 0 - 255 octets */
```
```suggestion
	case BT_AUDIO_METADATA_TYPE_PROGRAM_INFO: /* 0 - 255 octets */
```

Thalley commented:
@MariuszSkamra this fixes https://github.com/zephyrproject-rtos/zephyr/issues/59666 - Can you add it to the PR description? 

I was looking at the extended and vendor metadata as well. They are defined as 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/2520413/2b192801-6743-45e9-a994-da74dd6800dd)

While our checks are checking for `len < 1`. I think we can agree that the first octet shall always be present, but since the assigned numbers specifically mentions `Octet 2-254`, does that imply that the size of extended and vendor metadata shall be at least 1 (besides the type/company ID)? i.e. should the `len < 1` be `len < 2` for those?

MariuszSkamra commented:
> While our checks are checking for `len < 1`. I think we can agree that the first octet shall always be present, but since the assigned numbers specifically mentions `Octet 2-254`, does that imply that the size of extended and vendor metadata shall be at least 1 (besides the type/company ID)? i.e. should the `len < 1` be `len < 2` for those?

That's tricky but I think at least `Extended Metadata Type`/`Company_ID` should be there meaning the `Value` should be at least 2 bytes. I'll send PR for that

MariuszSkamra committed:
Bluetooth: audio: Fix metadata length checks

This fixes metadata length checks for BT_AUDIO_METADATA_TYPE_EXTENDED
and BT_AUDIO_METADATA_TYPE_VENDOR that should be at least 2 bytes long.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>
